helix-view/src/gutter.rs

```diff
@@ -64,25 +66,32 @@ pub fn diagnostic<'doc>(
             }
             use helix_core::diagnostic::Severity;
             if let Ok(index) = diagnostics.binary_search_by_key(&line, |d| d.line) {
```

Comment:
you can make the code below a lot simpler by using `partion_point` then you can avoid the closure (since you don't need to iterate backwards as partition_point always gives you the first index). I know that this is an existing issue but the code got more complex here so it would be nice to cut back on that.

Reply:
yeah looks good now, I had a minor comment (see below) but you can resolve this